application/client/src/app/ui/service/toolbar/chart.ts

```diff
@@ -0,0 +1,45 @@
+import { Observable, Subject } from 'rxjs';
+
+export interface IPositionChange {
+    left: number;
+    width: number;
+    full: number;
+}
+
+export class Chart {
```

Comment:
this class "lives" in the incorrect place. The chart isn't related to the toolbar. Actually chart "doesn't know" where is "he" would be rendered.